### PR TITLE
Change refund request flow

### PR DESCRIPTION
**Pull request by nuxzero**

### CODE REVIEW

File: src/main/java/co/omise/models/Refund.java [hunk 1]
```diff
@@ -56,20 +61,87 @@ public void setMetadata(Map<String, Object> metadata) {
...
+        }
+
+        public CreateRequestBuilder amount(long amount) {
             this.amount = amount;
             return this;
         }
 
-        public Create metadata(Map<String, Object> metadata) {
+        public CreateRequestBuilder metadata(Map<String, Object> metadata) {
             this.metadata = metadata;
```
Comment (Namiii) [inline src/main/java/co/omise/models/Refund.java]: Let's have both metadata(Map<> map) and metadata(String string, Object object) methods for this. Have a look at the way I've done it on the `Token` PR.

File: src/main/java/co/omise/resources/RefundResource.java [hunk 2]
```diff
@@ -22,23 +22,6 @@ public String chargeId() {
...
-    }
-
-    public Refund get(String refundId) throws IOException, OmiseException {
-        return httpGet(urlFor(refundId)).returns(Refund.class);
-    }
-
-    public Refund create(Refund.Create params) throws IOException, OmiseException {
-        return httpPost(urlFor("")).params(params).returns(Refund.class);
-    }
-
```
Comment (Namiii) [inline src/main/java/co/omise/resources/RefundResource.java]: Why not delete the whole class?

File: src/test/java/co/omise/live/BaseLiveTest.java [hunk 3]
```diff
@@ -6,11 +6,21 @@
...
     }
 
+    String getChargeId() {
+        return LIVETEST_CHARGE;
+    }
+
+    String getRefundId() {
+        return LIVETEST_REFUND;
+    }
+
```
Comment (Namiii) [inline src/test/java/co/omise/live/BaseLiveTest.java]: I feel these are a bit too specific to one test class for them to be in the `BaseLiveTest` class.

File: src/main/java/co/omise/resources/RefundResource.java (hunk 2 above)
Comment (nuxzero) [reply src/main/java/co/omise/resources/RefundResource.java]: @Namiii `ChargeSpecificResource` still using `RefundResource`. Not sure it safe to delete or not? 🤔

File: src/test/java/co/omise/live/BaseLiveTest.java (hunk 3 above)
Comment (nuxzero) [reply src/test/java/co/omise/live/BaseLiveTest.java]: Ok, I'll change this.

File: src/main/java/co/omise/models/Refund.java (hunk 1 above)
Comment (nuxzero) [reply src/main/java/co/omise/models/Refund.java]: 🤔 @pitiphong-p @chakrit Do we need both `metadata(Map<> map)` and `metadata(String string, Object object) `?

File: src/main/java/co/omise/resources/RefundResource.java (hunk 2 above)
Comment (Namiii) [reply src/main/java/co/omise/resources/RefundResource.java]: No you're right, we'll remove it once that resource is also removed 👍

File: src/main/java/co/omise/models/Refund.java (hunk 1 above)
Comment (chakrit) [reply src/main/java/co/omise/models/Refund.java]: Building maps is not that trivial in java. Requires at least 3 lines where the default case usually is just one key, one value ("invoice_id", "123") merchant ease of use wins here.

**Comment by nuxzero** [comment]
@Namiii I have done all your requested changes, please review again.

**Pull request merged**